### DISCUSSION TITLE
Allow ObservableProperty to be used in classes implementing INotifyPropertyChanged

### DISCUSSION THREAD

**Discussion by Moreno-Gentili** [CommunityToolkit/dotnet#659] [Ideas]
Hi everyone, suppose I liked my view models to be close to their view. So close, in fact, that I'd define my commands and properties in the code behind file (easier to understand, less files to manage).

```csharp
[ObservableObject]
public partial class MyClass : ContentView
{
    public MyClass()
    {
        InitializeComponent();
        BindingContext = this;
    }

    [ObservableProperty]
    int myProperty;

    [RelayCommand]
    void MyCommand()
    {
        MyProperty += 1;
    }
}
```

Unfortunately, this is not allowed. The error [MVVMTK0002](https://learn.microsoft.com/it-it/dotnet/communitytoolkit/mvvm/generators/errors/mvvmtk0002) is shown.
```
Cannot apply `[ObservableObject]` to type {type}, as it already declares the `INotifyPropertyChanged` interface.
```

What's exactly the problem here? Since a `ContentView` is already implementing `INotifyPropertyChanged`, the source generator should just add the missing parts.

**Comment by ewerspej** [comment]
Just a side note: When you place all commands and properties in your code-behind, then you're skipping the ViewModels and thus, you're not doing MVVM, because you're tight-coupling your business logic to the view. The name CommunityToolkit.Mvvm already suggests, that this is targeted at the Model-View-ViewModel pattern. I do agree that using Source Generators in code-behind files would be a valuable addition, I just wanted to point that out.

From a technical point of view, I think the problem with ContentViews and MVVM Source Generators is due to inheritance and partial classes. It should technically be possible to implement this, but the ContentViews should remain independent from MVVM architecture, meaning that concerns shouldn't be mixed. Therefore, the same Source Generators shouldn't be used for this.

**Comment by hansmbakker** [comment]
Apart from this not following the MVVM approach, the technical request is a duplicate of https://github.com/CommunityToolkit/dotnet/discussions/336
